common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -220,7 +220,10 @@ def non_editable_metadata_fields(self):
     @property
     def problem_types(self):
         """ Low-level problem type introspection for content libraries filtering by problem type """
-        tree = etree.XML(self.data)
+        try:
+            tree = etree.XML(self.data)
+        except etree.XMLSyntaxError:
+            return {}  # short-term fix to prevent errors (TNL-5057). Will be more properly addressed in TNL-4525.
         registered_tags = responsetypes.registry.registered_tags()
         return set([node.tag for node in tree.iter() if node.tag in registered_tags])
```

Comment:
Using a set comprehension here will save some space.  This creates a list, then a set, then throws away the list.